lib/generator.js

```diff
@@ -340,56 +338,67 @@ class Generator {
-              // that the package is already installed and it's a symbolic link.
-              const { resolvedLink } = await getLocalTemplateDetails(templatePath);
-              log.debug(`This template has already been installed and it's pointing to your filesystem at ${resolvedLink}.`);
-            }
-            installedPkg = require(path.resolve(templatePath, PACKAGE_JSON_FILENAME));
+            pkgPath = path.resolve(this.templateName);
+            installedPkg = require(path.resolve(pkgPath, PACKAGE_JSON_FILENAME));
+            
+            log.debug('Remember that your local template must have its own node_modules, "npm install" is not triggered.');
+          } else {
+            pkgPath = resolvePkg(this.templateName);
+            installedPkg = require(path.join(pkgPath, PACKAGE_JSON_FILENAME));
           }
+          packageVersion = installedPkg.version;
+          log.debug(`Template sources taken from ${pkgPath}.`);
+          if (packageVersion) log.debug(`Version of the template is ${packageVersion}.`);
 
           return resolve({
             name: installedPkg.name,
-            version: installedPkg.version,
-            path: path.resolve(DEFAULT_TEMPLATES_DIR, installedPkg.name),
+            path: pkgPath
           });
         } catch (e) {
+          log.debug(`Errors while trying to resolve package location at ${pkgPath}`, e);
           // We did our best. Proceed with installation...
         }
       }
 
-      // NOTE: npmi creates symbolic links inside DEFAULT_TEMPLATES_DIR
-      // (node_modules) for local packages, i.e., those located in the file system.
-      npmi({
-        name: this.templateName,
-        install: force,
-        path: ROOT_DIR,
-        pkgName: 'dummy value so it does not force installation always',
-        npmLoad: {
-          loglevel: 'http',
-          save: false,
-          audit: false,
-          progress: false,
-          only: 'prod'
-        },
-      }, (err, result) => {
-        if (err) return reject(err);
-        resolve(beautifyNpmiResult(result));
+      const debugMessage = force ? 'as you passed --install flag' : 'because the template cannot be found on disk';
+      log.debug(`Template installation started ${  debugMessage}`);
+
+      if (isFileSystemPath(this.templateName)) log.debug('Installation of template located on disk technically means symlink creation betweed node_modules of the generator and template sources. Your local template must have its own node_modules, "npm install" is not triggered.');
+
+      const arb = new Arborist({
+        path: ROOT_DIR
       });
+
+      try {
+        const installResult = await arb.reify({
+          add: [this.templateName],
+          saveType: 'prod',
+          save: false
+        });
+        //trying to figure out if this is the right way to access the name of added package https://github.com/npm/arborist/issues/243
```

Comment:
Does this sound like a temporary comment? 🤔

Reply:
yeap, they never replied 🤷🏼‍♂️ 
I will keep my issue open and if they write something important I will work on a new PR

comment removed